### ISSUE TITLE
pr_58021 caused a decrease in the accuracy of executing mkldnn inference for model det_mv3_east

### ISSUE THREAD

**Issue by EmmonsCurse** [PaddlePaddle/Paddle#58050]
### bug描述 Describe the Bug

**Error Description：** After the merge of https://github.com/PaddlePaddle/Paddle/pull/58021 the MKL-DNN inference accuracy of the det_mv3_east has decreased.

The max_diff comparison:

- test_det_mv3_east_mkldnn.py 
   before(oneDNN v3.1.1)：6.600469e-05 
   after(oneDNN v3.2.1): 151.867

**1. Operating environment**
PaddlePaddle version：develop
OS version：CentOS
Python version：3.8

**2. The detailed error information: (The comparison strategy is to exit when the first one is greater than the diff threshold)**
```
test_det_mv3_east_mkldnn.py:54:
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

self = <test_mkldnn_helper.TestModelInferenceMKLDNN object at 0x7f3f5f4f1d60>
result = array([-16.915035 , -12.312555 , -13.390684 , ...,   0.2960205,
        -0.9806156,  -4.6984196], dtype=float32)
expect = array([-13.676786 ,  -3.9301395,  -7.5552464, ...,  -3.3748627,
        -4.0334225,  -4.0430546], dtype=float32)
delta = 0.001

    def check_data(self, result, expect, delta):
        """
        check result
        Args:
            result(list): list of result data
            expect(list): list of expect data
            delta(float): e.g. 0.001
        Returns:
            None
        """
        logger.info("current comparison delta is : {0}".format(delta))
        assert len(result) == pytest.approx(len(expect)), "output length not equal"
        for i in range(0, len(expect)):
>           assert result[i] == pytest.approx(expect[i], rel=None, abs=delta), "output data not equal"
E           AssertionError: output data not equal
E           assert -16.915035 == -13.676786422729492 ± 1.0e-03
E             comparison failed
E             Obtained: -16.915035247802734
E             Expected: -13.676786422729492 ± 1.0e-03

test_mkldnn_helper.py:60: AssertionError
```
**3. The problem recurrence steps：**
```
git clone https://github.com/EmmonsCurse/continuous_integration.git -b rename_fluid_to_base --depth=1

cd ./continuous_integration/inference/inference_api_test/python_api_test/

project_path=`pwd`
export project_path
cd ${project_path}

# download Data

mkdir -p ./Data
cd ./Data

# download models

wget --no-proxy -q https://sys-p0.bj.bcebos.com/inference/python-ocr-infer.tgz --no-check-certificate
tar -xvf python-ocr-infer.tgz

cd -

# requirements
python -m pip install --upgrade pip -i https://mirror.baidu.com/pypi/simple
python -m pip install -r requirements.txt -i https://mirror.baidu.com/pypi/simple

# download paddlepaddle_whl

## error_pr(failed)
#wget -q https://paddle-qa.bj.bcebos.com/paddle-pipeline/Develop-GpuAll-LinuxCentos-Gcc82-Cuda112-Trtoff-Py38-Compile/951e539ba3504e52e4979bae8a03d336487c3208/paddlepaddle_gpu-0.0.0-cp38-cp38-linux_x86_64.whl

## the previous commit of error_pr(passed)
#wget -q https://paddle-qa.bj.bcebos.com/paddle-pipeline/Develop-GpuAll-LinuxCentos-Gcc82-Cuda112-Trtoff-Py38-Compile/2f56c8e1a4a1eb3033cb2f02ce16f85346a43b09/paddlepaddle_gpu-0.0.0-cp38-cp38-linux_x86_64.whl

# install paddlepaddle_whl
python -m pip install paddlepaddle_gpu-0.0.0-cp38-cp38-linux_x86_64.whl

# run case
cd ./tests/mkldnn

python -m pytest test_det_mv3_east_mkldnn.py
```

**note**
It is also positioned that the reason of this problem mainly as the impact of an conv_transpose_bn_fuse_pass. There's a temporary solution that you can add 'predictor_config.delete_pass('conv_transpose_bn_fuse_pass') ' below continuous_integration/inference/inference_api_test/python_api_test/src/test_case.py line 102 https://github.com/EmmonsCurse/continuous_integration/blob/rename_fluid_to_base/inference/inference_api_test/python_api_test/src/test_case.py#L102C63-L102C63

```
elif config_type == 'mkldnn':
     predictor_config.disable_gpu()
     predictor_config.enable_mkldnn()
     predictor_config.set_cpu_math_library_num_threads(4)
     predictor_config.delete_pass('conv_transpose_bn_fuse_pass')
```

### 其他补充信息 Additional Supplementary Information

@zhanglirong1999 @xinyu-intel Can you help me solve this problem?

**Comment by xinyu-intel** [comment]
Hi @EmmonsCurse , thank you for reporting the issue. We'll investigate on this issue. cc @LLee233

**Comment by LLee233** [comment]
Hi @EmmonsCurse , just want to make sure what kind of machine you are using? Could you provide us some info like specific type of CPU/GPU?

**Comment by EmmonsCurse** [comment]
> Hi @EmmonsCurse , just want to make sure what kind of machine you are using? Could you provide us some info like specific type of CPU/GPU?

Hi @LLee233 The detailed information of my machine's OS CPU and GPU is as follows:
```
OS: 
Linux 3 3.10.0-1062.18.1.el7.x86_64 #1 SMP Tue Mar 17 23:49:17 UTC 2020 x86_64 x86_64 x86_64 GNU/Linux

CPU: 
vendor_id	: GenuineIntel
cpu family	: 6
model		: 85
model name	: Intel(R) Xeon(R) Gold 6271C CPU @ 2.60GHz
stepping	: 7
microcode	: 0x1
cpu MHz		: 2600.000
cache size	: 33792 KB
physical id	: 1
siblings	: 40
core id		: 19
cpu cores	: 20
apicid		: 103
initial apicid	: 103
fpu		: yes
fpu_exception	: yes
cpuid level	: 13
wp		: yes
flags		: fpu vme de pse tsc msr pae mce cx8 apic sep mtrr pge mca cmov pat pse36 clflush mmx fxsr sse sse2 ss ht syscall nx pdpe1gb rdtscp lm constant_tsc arch_perfmon rep_good nopl xtopology eagerfpu pni pclmulqdq monitor ssse3 fma cx16 pcid sse4_1 sse4_2 x2apic movbe popcnt tsc_deadline_timer aes xsave avx f16c rdrand hypervisor lahf_lm abm 3dnowprefetch invpcid_single ssbd ibrs ibpb fsgsbase tsc_adjust bmi1 avx2 smep bmi2 erms invpcid mpx avx512f avx512dq rdseed adx smap clflushopt clwb avx512cd avx512bw avx512vl xsaveopt xsavec xgetbv1 arat umip pku ospke avx512_vnni spec_ctrl
bogomips	: 5200.00
clflush size	: 64
cache_alignment	: 64
address sizes	: 46 bits physical, 48 bits virtual

GPU:
GPU 0: Tesla T4 (UUID: GPU-b288f3b1-121a-2a96-781d-1339e3180789)
GPU 1: Tesla T4 (UUID: GPU-339a60af-81ea-1af5-775b-8cbb84d94578)
GPU 2: Tesla T4 (UUID: GPU-23de26e6-b676-b1cc-f590-be89c12a8bf0)
GPU 3: Tesla T4 (UUID: GPU-938c87f8-5452-c6aa-4176-119945efcf5c)
```

**Comment by EmmonsCurse** [comment]
@xinyu-intel @LLee233  Hi, there is another question of [ pr_58021](https://github.com/PaddlePaddle/Paddle/pull/58021/files#:~:text=if(WIN32,WITH_MKLDNN%20OFF)) that why set WITH_MKLDNN=OFF when MSVC_VERSION LESS 1920 in WIN32?

**Comment by zhanglirong1999** [comment]
> @xinyu-intel @LLee233 Hi, there is another question of [ pr_58021](https://github.com/PaddlePaddle/Paddle/pull/58021/files#:~:text=if(WIN32,WITH_MKLDNN%20OFF)) that why set WITH_MKLDNN=OFF when MSVC_VERSION LESS 1920 in WIN32?

Since oneDNN later version has deprecated support for VS2017 and claim the support for VS2019 and VS2022

**Comment by EmmonsCurse** [comment]
> > @xinyu-intel @LLee233 Hi, there is another question of [ pr_58021](https://github.com/PaddlePaddle/Paddle/pull/58021/files#:~:text=if(WIN32,WITH_MKLDNN%20OFF)) that why set WITH_MKLDNN=OFF when MSVC_VERSION LESS 1920 in WIN32?
> 
> Since oneDNN later version has deprecated support for VS2017 and claim the support for VS2019 and VS2022

@zhanglirong1999 Thanks for reply, thus, I suggest adding some explanation about this setting in the PR description.

**Comment by LLee233** [comment]
Hi @EmmonsCurse , we've created a PR for fixing this, you may try with it [#58181](https://github.com/PaddlePaddle/Paddle/pull/58181) and see if it works, thanks.

**Comment by EmmonsCurse** [comment]
> Hi @EmmonsCurse , we've created a PR for fixing this, you may try with it [#58181](https://github.com/PaddlePaddle/Paddle/pull/58181) and see if it works, thanks.

@LLee233 Sorry, I'm currently verifying and the results will be given later. Thanks~

**Comment by EmmonsCurse** [comment]
> > Hi @EmmonsCurse , we've created a PR for fixing this, you may try with it [#58181](https://github.com/PaddlePaddle/Paddle/pull/58181) and see if it works, thanks.
> 
> @LLee233 Sorry, I'm currently verifying and the results will be given later. Thanks~

@LLee233 Thank you, the problem has been fixed by #58181.

**Comment by xinyu-intel** [comment]
@EmmonsCurse @LLee233 Thank you for the reporting and fixing. The fix has been merged.
